src/index.js

```diff
@@ -2,15 +2,32 @@ import React from 'react';
 import ReactDOM from 'react-dom/client';
 import './index.css';
 import { BrowserRouter, Routes, Route } from 'react-router-dom';
+import { Provider } from 'react-redux';
 import App from './App';
-import Categories from './routes/Categories';
+import Categories from './components/categories/Categories';
+import store from './redux/configureStore';
 
 const root = ReactDOM.createRoot(document.getElementById('root'));
 root.render(
   <BrowserRouter>
     <Routes>
-      <Route path="/" element={<App />} />
-      <Route path="categories" element={<Categories />} />
+      <Route
+        path="/"
+        element={(
+          <Provider store={store}>
+            <App />
+          </Provider>
+              )}
+      />
+
+      <Route
+        path="categories"
+        element={(
+          <Provider store={store}>
+            <Categories />
+          </Provider>
+       )}
+      />
     </Routes>
   </BrowserRouter>,
```

Comment:
- [ ] [OPTIONAL] Please make the store only wrap the `BrowserRouter` and it will be provided to the whole application instead of wrapping each component with it so it will be like:
```
<Provider store={store}>
  <BrowserRouter>
    <Routes>
             ....
    </Routes>
  </BrowserRouter>
</Provider>
```

Reply:
Hi @gemmen29 ,

Thanks you so much for that insight. I did not know that's possible